src/provisioner/log.py

```diff
@@ -25,23 +27,47 @@ class CortxProvisionerLog(Log):
     logger = None
 
     @staticmethod
-    def init(service_name, log_path=None, level='INFO', console_output=False):
+    def initialize(service_name, log_path=const.DEFAULT_LOG_PATH, level='INFO', console_output=True):
         """
         Initialize and use cortx-utils logger to log message in file and console.
         If console_output is True, log message will be displayed in console.
         """
+        if not CortxProvisionerLog.logger:
+            log_path = os.path.join(log_path, const.APP_NAME)
+            level = level if level in const.SUPPORTED_LOG_LEVELS else 'INFO'
+            Log.init(service_name, log_path, level=level, console_output=console_output)
+            CortxProvisionerLog.logger = Log.logger
 
-        if not log_path:
-            log_path = os.path.join('/var/log/cortx', 'provisioner')
 
-        if level not in ['DEBUG', 'INFO', 'WARNING', 'ERROR', 'CRITICAL']:
-            level = 'INFO'
+    @staticmethod
+    def reinitialize(service_name, log_path=const.DEFAULT_LOG_PATH, level='INFO', console_output=True):
+        """
+        Reinitialize existing logger.
+
+        This removes logging handler from existing logger and moves captured
+        logs from temporary log file to target log path file.
+        """
+        if Log.logger:
+            log_path = os.path.join(log_path, const.APP_NAME)
+            level = level if level in const.SUPPORTED_LOG_LEVELS else 'INFO'
 
-        Log.init(service_name, log_path, level=level, console_output=console_output)
-        CortxProvisionerLog.logger = Log.logger
+            # Remove current logging handlers before truncating
+            for handler in Log.logger.handlers[:]:
+                Log.logger.removeHandler(handler)
 
+            temp_log_file = '%s/%s/%s.log' % (
```

Comment:
Lets put some date-timestamp for the log file, else we would just overwrite existing file...

Reply:
Yes, overwrite is the expectation here. Let me explain it better,

This tmp_log_file will be created well before cortx_setup command process.
Logs will be appended to this file until reinitialize() called.
Once reinitialize() called, content of the tmp_log_file will be moved out. So we will have empty tmp_log_file for next cycle.